index.ts

```diff
@@ -117,28 +91,17 @@ function _utf8_encode(string: string) {
  * @private
  * @param {string} utftext - UTF-8 text to dencode
  */
-function _utf8_decode(utftext: string) {
-  let string = '';
-  let i = 0;
-
-  while (i < utftext.length) {
-    const c = utftext.charCodeAt(i);
-
-    if (c < 128) {
-      string += String.fromCharCode(c);
-      i++;
-    } else if (c > 191 && c < 224) {
-      const c2 = utftext.charCodeAt(i + 1);
-      string += String.fromCharCode(((c & 31) << 6) | (c2 & 63));
-      i += 2;
-    } else {
-      const c2 = utftext.charCodeAt(i + 1);
-      const c3 = utftext.charCodeAt(i + 2);
-      string += String.fromCharCode(
-        ((c & 15) << 12) | ((c2 & 63) << 6) | (c3 & 63),
-      );
-      i += 3;
-    }
+function _utf8_decode(str: string): string {
+  try {
+    return decodeURIComponent(
+      str
+        .split('')
+        .map((c) => '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2))
+        .join('')
+    );
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error('Failed to decode utf-8 base64 string: ', error);
+    return str;
```

Comment:
IMO this library is so tiny (it's a single util) that it's better to just throw an error when it happens. At least user can catch the error to react on it.
Returning `str` might lead to unhandled errors - when user expects that the string was decoded whilst in fact, it was not. With your implementation, there is simply no way to tell from the outside code the string was not decoded.

I'd recommend removing try/catch block whatsoever

Reply:
I accept that and just removed the try/catch block, and I noticed that my ESLint plugin is not working, sorry for the lint issue and they has been fixed too :D.